source/sundials/kinsol.cc

```diff
@@ -338,24 +492,24 @@ namespace SUNDIALS
   template class KINSOL<Vector<double>>;
   template class KINSOL<BlockVector<double>>;
 
-#    ifdef DEAL_II_WITH_MPI
+#  ifdef DEAL_II_WITH_MPI
 
-#      ifdef DEAL_II_WITH_TRILINOS
+#    ifdef DEAL_II_WITH_TRILINOS
   template class KINSOL<TrilinosWrappers::MPI::Vector>;
   template class KINSOL<TrilinosWrappers::MPI::BlockVector>;
-#      endif
+#    endif
 
-#      ifdef DEAL_II_WITH_PETSC
-#        ifndef PETSC_USE_COMPLEX
+#    ifdef DEAL_II_WITH_PETSC
+#      ifndef PETSC_USE_COMPLEX
```

Comment:
Can we perhaps instantiate the KINSOL class on all `VECTOR_TYPES` or `REAL_VECTOR_TYPES`?

Reply:
I thought about that but in the end decided against it. We'd need to write the `copy()` functions for all vector types, and I don't really know how to address parallel computations. Let me defer this to a later time if you don't ind!

Reply:
Or just use `NVectorView` instead of `copy` which works for all vector types;) (although it might not have all operations required by KINSOL?)

Reply:
Yes, that's the long-term goal.

Reply:
>  Let me defer this to a later time if you don't mind!

Yip, no problem. Also, if there's a better solution (`NVectorView`) then I'd support us adopting that instead.

Reply:
Yes, the `NVectorView` thing is the way to go. It would allow us to keep using the vector we have, rather than copying stuff back and forth. I just need to learn how to use it -- see #11395.